src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriBuilder.java

```diff
@@ -72,6 +72,132 @@
      */
     private final static Logger logger = Logger.getLogger(ColibriBuilder.class);
 
+    /**
+     * Copies the transport info from a {@link ContentPacketExtension} to a
+     * {@link ColibriConferenceIQ.ChannelCommon}.
+     * @param content the {@link ContentPacketExtension} from which to copy.
+     * @param channel the {@link ColibriConferenceIQ.ChannelCommon} to which to copy.
+     */
+    private static void copyTransport(
+        ContentPacketExtension content,
+        ColibriConferenceIQ.ChannelCommon channel)
+    {
+        IceUdpTransportPacketExtension transport
+            = content.getFirstChildOfType(
+            IceUdpTransportPacketExtension.class);
+
+        channel.setTransport(
+            IceUdpTransportPacketExtension
+                .cloneTransportAndCandidates(transport, true));
+    }
+
+    /**
+     * Copies the contents of the description element from a
+     * {@link ContentPacketExtension} to a {@link ColibriConferenceIQ.Channel}.
+     * @param content the {@link ContentPacketExtension} from which to copy.
+     * @param channel the {@link ColibriConferenceIQ.Channel} to which to copy.
+     */
+    private static boolean copyDescription(
+        ContentPacketExtension content,
+        ColibriConferenceIQ.Channel channel)
+    {
+        RtpDescriptionPacketExtension description
+            = content.getFirstChildOfType(
+            RtpDescriptionPacketExtension.class);
+        if (description != null)
+        {
+            return copyDescription(description, channel);
+        }
+        return false;
+    }
+
+    /**
+     * Copies the contents of a {@link RtpDescriptionPacketExtension} to a
+     * {@link ColibriConferenceIQ.Channel}.
+     * @param description the {@link RtpDescriptionPacketExtension} from which
+     * to copy.
+     * @param channel the {@link ColibriConferenceIQ.Channel} to which to copy.
+     */
+    private static boolean copyDescription(
```

Comment:
would this be better as a clone method inside of RtpDescriptionPacketExtension?

Reply:
after reading this some more, i'm wondering if 'copy' feels accurate...we don't actually make a copy of the description, more extract fields from it to put in the channel?  is this something more like `parseRtpDescriptionToChannel` or `addPayloadTypesAndHdrExtensions`?

Reply:
copyDescriptionToChannel?

Reply:
my only hesitation with 'copy' was that we don't actually do a deep copy of the header extensions (just add the references), but if you don't think that's a big deal then i'm fine with `copyDescriptionToChannel`